src/openlifu/photoscan.py

```diff
@@ -251,10 +253,67 @@ def run_reconstruction(images: list[Path], pipeline: Path | None = None) -> Phot
 
     subprocess.run(command, check=True)
 
-    photoscan_dict = {"model_filename": str(output_dir / "texturedMesh.obj"),
-     "texture_filename": str(output_dir / "texture_1001.png"),
-     "mtl_filename": str(output_dir / "texturedMesh.mtl")}
+    output_dir_merged = temp_dir / "output_dir_merged"
+    output_dir_merged.mkdir(parents=True, exist_ok=True)
+
+    merge_textures(output_dir / "texturedMesh.obj", output_dir_merged)
+
+    photoscan_dict = {"model_filename": str(output_dir_merged / "texturedMesh.obj"),
+     "texture_filename": str(output_dir_merged / "material_0.png"),
+     "mtl_filename": str(output_dir_merged / "material.mtl")}
```

Comment:
There's a small problem here, not related to the texture merging but just a quirk of the `Photoscan` class.

The `texture_filename` and `mtl_filename` are meant to be filenames, not full paths.

So to make this usable, the `run_reconstruction` function should return the absolute filepaths separately in addition to the `Photoscan` object. It may seem weird but it's because of how the "database" works.

Reply:
(I made the fix)